extractor/extractor.py

```diff
@@ -0,0 +1,205 @@
+import re
+from urllib.request import urlopen
+from pprint import pprint
+
+import requests
+import bs4
+from bs4 import BeautifulSoup, Comment
+from PIL import Image
+
+import contents
+
+req_session = ""
+
+
+class Extractor:
+    def __init__(self, URL):
+        self.url = URL
+        self.soup = ''
+        self.contents_list = contents.Contents()
+
+    def parse(self):
+        global req_session
+
+        req_session = requests.Session()
+        file = req_session.get(self.url).text    # To request Html from URL
+
+        # Remove the comment to read local files for testing purpose
+        # file = open('./test_html.html','r').read()
+
+        self.soup = BeautifulSoup(file,'lxml')
+        self.clean_soup()
+
+        self.extract()
+
+    def clean_soup(self):
+        decomposable_tags = self.soup.find_all(['script', 'noscript', 'style', 'header', 'footer', 'meta', 'link'])
+        comments = self.soup.find_all(text=lambda text: isinstance(text, Comment))
+        ads = self.soup.find_all(re.compile('.*'), {'class': re.compile('(^ad-|^ads-)')})
+
+        comments = [comment.extract() for comment in comments]
+        [tag.decompose() for tag in decomposable_tags]
+        [ad.decompose() for ad in ads]
+
+    def extract(self):
+        # Extract data from head
+        head = self.soup.find('head')
+        self.extract_title_tag(head)
+
+        # Extract data from body
+        body_soup = self.soup.find('body')
+        self.iterate_over_dom(body_soup)
+
+    def extract_title_tag(self, soup_head):
+        title = soup_head.title
+        _content = contents._Content()
+        _content.contentType = 'Text'
+        _content.text.type = 'title'
+        _content.text.text_string = title.get_text()
```

Comment:
It would be better if everything is passed to ctor, instead of creating and setting properties.